core/domain/caching_services_test.py

```diff
@@ -0,0 +1,368 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for methods in core.domain.caching_services"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import caching_domain
+from core.domain import caching_services
+from core.domain import exp_domain
+from core.tests import test_utils
+
+import python_utils
+
+
+class CachingServicesUnitTests(test_utils.GenericTestBase):
+    """Tests for caching_services."""
+
+    def test_retrieved_memory_profile_contains_correct_elements(self):
+        memory_profile = caching_services.get_memory_cache_stats()
+        self.assertIsInstance(memory_profile, caching_domain.MemoryCacheStats)
+        self.assertIsNotNone(memory_profile.total_allocated_in_bytes)
+        self.assertIsNotNone(memory_profile.peak_memory_usage_in_bytes)
+        self.assertIsNotNone(memory_profile.total_number_of_keys_stored)
+
+    def test_flush_cache_wipes_cache_clean(self):
+        """Tests whether flushing the cache removes the elements in the
+        cache.
+        """
+        key_value_mapping = {'a': '1', 'b': '2', 'c': '3'}
+        caching_services.set_multi(
+            caching_services.CACHE_NAMESPACE_DEFAULT, None, key_value_mapping)
+        exploration_id = 'id'
+        default_exploration = (
+            exp_domain.Exploration.create_default_exploration(
+                'exp_id_1', title='A title', category='A category'))
+        caching_services.set_multi(
+            caching_services.CACHE_NAMESPACE_EXPLORATION,
+            python_utils.convert_to_bytes(0),
```

Comment:
Here and elsewhere, why not just use '0'? Having convert_to_bytes() everywhere makes the test harder to read.